common/src/com/thoughtworks/go/config/materials/Materials.java

```diff
@@ -142,7 +136,7 @@ public void cleanUp(File baseFolder, ConsoleOutputStreamConsumer consumer) {
 
     boolean hasOneMaterialUseBaseFolder() {
         for (Material material : this) {
-            if (material.getFolder() == null) {
+            if (((AbstractMaterial) material).requiresDestinationFolder() && !((AbstractMaterial) material).hasDestinationFolder()) {
```

Comment:
The two checks cannot be combined. For a pipeline & scm material with a destination folder we don't want to return true.